SU2_CFD/include/solvers/CScalarSolver.inl

```diff
@@ -832,3 +832,36 @@ void CScalarSolver<VariableType>::SetResidual_DualTime(CGeometry* geometry, CSol
 
   }  // end dynamic grid
 }
+
+
+template <class VariableType>
+void CScalarSolver<VariableType>::PushSolutionBackInTime(unsigned long TimeIter, bool restart,CSolver*** solver_container,
+                              CGeometry** geometry, CConfig* config) {
+  const bool dual_time = config->GetTime_Marching() == TIME_MARCHING::DT_STEPPING_1ST ||
+                         config->GetTime_Marching() == TIME_MARCHING::DT_STEPPING_2ND;
+  const bool isRestartIter = restart && TimeIter == config->GetRestart_Iter();
+
+  /*--- The value of the solution for the first iteration of the dual time. ---*/
+
+  if (dual_time && (TimeIter == 0 || isRestartIter)) {
+    /*--- Push back the initial condition to previous solution containers
+     for a 1st-order restart or when simply intitializing to freestream. ---*/
+
+    for (auto iMesh = 0u; iMesh <= config->GetnMGLevels(); iMesh++) {
+      nodes->Set_Solution_time_n();
+      nodes->Set_Solution_time_n1();
+    }
```

Comment:
This doesn't look right, the heat solver does this on the nodes of the coarse solvers.
If you don't have MG for species, remove the loop over grids.